packages/fx-core/src/common/tools.ts

```diff
@@ -284,7 +284,12 @@ export async function saveFilesRecursively(
   await Promise.all(
     zip
       .getEntries()
-      .filter((entry) => !entry.isDirectory && entry.entryName.includes(appFolder))
+      .filter(
+        (entry) =>
+          !entry.isDirectory &&
+          entry.entryName.includes(appFolder) &&
+          entry.entryName.split("/").includes(appFolder)
+      )
```

Comment:
Suppose the following scenario:
1. There is one sample named `sample-1`
2. Ther is another sample named `sample-2`.
3. In `sample-2`, there is a file accidentally named `sample-1`, like `sample-2/some/path/sample-1`

If user downloads `sample-1`, will `sample-2/some/path/sample-1` be also downloaded?

Reply:
Good catch. That's correct. But there's not an ideal way to check which one belongs to current app folder. 

One way I can think of is to check the second string after splitting using '/' but actually we can't make sure the folder structure is fixed either. So since the scenario you mentioned should have little possibility. I think we can fix it in this way for now.

Reply:
Understood. Agree to just quick fix the bug introduced by `todo-list-with-Azure-backend-M365` sample.